RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -61,27 +61,97 @@ public bool Generate(ProcessedArgs args)
+                    foreach (IPackage dependencyPackage in dependencyPackages)
+                    {
+                        bool dependencyPackageHasAnalyzers = HasAnalyzers(dependencyPackage, args.Language);
+
+                        if (!dependencyPackageHasAnalyzers)
+                        {
+                            this.logger.LogWarning(UIResources.APG_NoAnalyzersFound, dependencyPackage.Id);
+                        }
+
+                        dependencyPackagesHaveAnalyzers = dependencyPackagesHaveAnalyzers || dependencyPackageHasAnalyzers;
+                    }
+
+                    if (!dependencyPackagesHaveAnalyzers)
+                    {
+                        return false;
+                    }
+                }
             }
 
-            IEnumerable<IPackage> licenseAcceptancePackages = this.GetPackagesRequiringLicenseAcceptance(package);
-            if (licenseAcceptancePackages.Any() && ! args.AcceptLicenses)
+            // Check for packages that require the user to accept a license
+            IEnumerable<IPackage> licenseAcceptancePackages = this.GetPackagesRequiringLicenseAcceptance(targetPackage);
+            if (licenseAcceptancePackages.Any() && !args.AcceptLicenses)
             {
-                this.logger.LogError(UIResources.APG_NGPackageRequiresLicenseAcceptance, package.Id, package.Version.ToString());
+                this.logger.LogError(UIResources.APG_NGPackageRequiresLicenseAcceptance, targetPackage.Id, targetPackage.Version.ToString());
                 this.ListPackagesRequiringLicenseAcceptance(licenseAcceptancePackages);
                 return false;
             }
 
+            // Initial run with the user-targeted package and arguments
+            bool result = GeneratePluginForPackage(args, targetPackage);
+            if (!result)
+            {
+                return false;
+            }
+
+            // Dependent package generation changes the arguments
+            if (args.RecurseDependencies)
+            {
+                this.logger.LogWarning(UIResources.APG_RecurseEnabled_SQALENotEnabled);
+
+                foreach (IPackage currentPackage in dependencyPackages)
+                {
+                    // No way to specify the SQALE file for any but the user-targeted package at this time
+                    ProcessedArgs currentArgs = new ProcessedArgs(currentPackage.Id, currentPackage.Version, 
+                        args.Language, null, args.AcceptLicenses, args.RecurseDependencies, args.OutputDirectory);
+
+                    result = GeneratePluginForPackage(currentArgs, currentPackage);
+                    if (!result)
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            LogAcceptedPackageLicenses(licenseAcceptancePackages);
+
+            return true;
+        }
+
+        private bool GeneratePluginForPackage(ProcessedArgs args, IPackage package)
```

Comment:
Previously it did just mean "a jar has been created" but that isn't the case now due to line 148 - the function now returns _true_ in two different cases. If it's not obvious what will be returned then it is better to document it.
Having said that, this problem should go away if you only pass in packages that contain analysers.